flocker/node/agents/test/test_blockdevice.py

```diff
@@ -146,6 +154,235 @@ def test_only_unattached_devices(self):
         self.assertDiscoveredState(self.deployer, [])
 
 
+class BlockDeviceDeployerDestructionCalculateNecessaryStateChangesTests(
+        SynchronousTestCase
+):
+    """
+    Tests for ``BlockDeviceDeployer.calculate_necessary_state_changes``
+    in the cases relating to dataset destruction.
+    """
+    def test_undeleted_dataset_not_deleted(self):
+        """
+        ``BlockDeviceDeployer.calculate_necessary_state_changes`` does not
+        calculate a change to destroy datasets that are not marked as deleted
+        in the configuration.
+        """
+        dataset_id = uuid4()
+        node = u"192.0.2.1"
+        local_state = NodeState(
+            hostname=node,
+            manifestations={
+                Manifestation(
+                    dataset=Dataset(
+                        dataset_id=unicode(dataset_id),
+                    ),
+                    primary=True,
+                ),
+            },
+            paths={
+                unicode(dataset_id):
+                FilePath(b"/flocker/").child(bytes(dataset_id)),
+            },
+        )
+        cluster_state = Deployment(
+            nodes={local_state.to_node()}
+        )
+
+        local_config = local_state.to_node()
```

Comment:
This is going to fail; merge forward and maybe reuse the `to_node()` test utility I wrote for `test_loop.py`.